compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -388,33 +388,32 @@ fn try_execute_query<CTX, C>(
     span: Span,
     key: C::Key,
     lookup: QueryLookup<'_, CTX::DepKind, CTX::Query, C::Key, C::Sharded>,
+    dep_node: Option<DepNode<CTX::DepKind>>,
```

Comment:
There's a lot going on in 6f001c7c204a33c0b8907f98e8f5d1211bd7c9b7. Can you leave some comments about what the goal of the changes in this commit are?

Reply:
I added a few comments in the commit message, and split a few parts into other commits.